### ISSUE TITLE
Filtering does not work for a full workbook (--fullpdf)

### ISSUE THREAD

**Issue by jacalata** [tableau/tabcmd#233]
Filtering the data in a viz for export only works when exporting a single view at a time. 

tabcmd export /WorldIndicators/Population --pdf  -f b1.pdf --filter "Birth Rate Bin=1.5-3%"

https://github.com/tableau/tabcmd/issues/176#issuecomment-1451113454_

**Comment by graham-marcus** [comment]
Noting the same issue
tabcmd export "WorkbookName/ViewName?Hotel%20Code=ADLAD" --pdf -f "C:\temp\filename ADLAD.pdf"
Filters correctly
tabcmd export "WorkbookName/ViewName?Hotel%20Code=ADLAD" --fullpdf -f "C:\temp\filename.pdf"
does not pass the filter

**Comment by KKriplani** [comment]
Hi All,
Is there any update to this issue?

Basically, we won't be able to print a pdf file with multiple pages and a filter until this issue is resolved, right?
If anyone knows of a work around to print them into one pdf, please let me know.

Thanks in advance.

Regards,
KKriplani

**Comment by kenpedersen** [comment]
Any update on this issue affecting filtering --fullpdf exports, or any known workarounds aside from splitting the views into separate PDFs? The issue still persists in v2.0.11.

Thank you.

**Comment by TimLafferty** [comment]
This is a HUGE problem. I generate monthly billing statements for 50k+ users via TabCmd. Any updates?

**Comment by BGirkins** [comment]
I can confirm the same issue. Very frustrating to force tabcmd 2.0 and not even have the same level of functionality. No communication on a potential fix either.

**Comment by smesecurity999329324442** [comment]
Well. Here we are 3 months later and the issue still persists. Is anyone at their desk?

**Comment by kenpedersen** [comment]
FYI below is the response I got after opening a support case. We have moved away from TabCMD due to this issue, and are instead calling the Query View PDF endpoint from Python.

> TabCMD 2.0 is one of Tableau Open-Source Projects, meaning that it is only supported via the GitHub site. So, no support is available from Tableau Technical Support.
>  
> For the particular issue of Filters not being applied when attempting to download workbooks as a PDF, that is the result of a current limitation with the Tableau Server REST API, rather than a bug in TabCMD. Since TabCMD uses the REST API to interact with Tableau Server/Cloud it is limited by the REST API and currently the REST API doesn't support filtering when downloading a workbooks as a PDF. Filtering is supported on Query View PDF.
>  
> Download Workbook PDF
> [https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#download_workbook_pdf](url)
>  
> Query View PDF
> [https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#query_view_pdf](url)
>  
> If you require Filtering for Workbook PDF Download, I suggest using the Original TabCMD.
>  
> Automating PDF Exports with Filters Using Tabcmd
> [https://kb.tableau.com/articles/HowTo/automating-pdf-exports-with-filters-using-tabcmd](url)
>  
> There is an internal feature request to add this functionality to the REST API and I have added this case to it.  Another option to provide suggestions for product enhancements is to submit an idea through our community: [https://community.tableau.com/s/ideas](url) Providing feedback through the community forums allows other users to vote and discuss feature requests and allows our development team to gauge the demand for each enhancement. Your feedback is invaluable and helps us improve our software.
>  
> Since the enhancement request has been submitted on your behalf, I will be moving this case into a closed status.

**Comment by jacalata** [comment]
That's a somewhat confused response from support and I will follow up to make sure they understand that they do, in fact, offer support on both versions of tabcmd.

However, they are correct that the missing piece here is functionality *on the server*. There is no code update to Tabcmd 2 that will make workbook filtering work, it needs to be shipped in an update to server. I don't know a timeline of when that will happen. 

A couple notes
- I am not sure why using the Query View PDF call would help you, because that's exactly what tabcmd calls to return you a pdf. 
- if this is breaking your workflow, then it is grounds for an MFA exception until fixed, so that you can continue using old tabcmd.

**Comment by BGirkins** [comment]
> FYI below is the response I got after opening a support case. We have moved away from TabCMD due to this issue, and are instead calling the Query View PDF endpoint from Python.
> 
> > TabCMD 2.0 is one of Tableau Open-Source Projects, meaning that it is only supported via the GitHub site. So, no support is available from Tableau Technical Support.
> > For the particular issue of Filters not being applied when attempting to download workbooks as a PDF, that is the result of a current limitation with the Tableau Server REST API, rather than a bug in TabCMD. Since TabCMD uses the REST API to interact with Tableau Server/Cloud it is limited by the REST API and currently the REST API doesn't support filtering when downloading a workbooks as a PDF. Filtering is supported on Query View PDF.
> > Download Workbook PDF
> > [https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#download_workbook_pdf](url)
> > Query View PDF
> > [https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#query_view_pdf](url)
> > If you require Filtering for Workbook PDF Download, I suggest using the Original TabCMD.
> > Automating PDF Exports with Filters Using Tabcmd
> > [https://kb.tableau.com/articles/HowTo/automating-pdf-exports-with-filters-using-tabcmd](url)
> > There is an internal feature request to add this functionality to the REST API and I have added this case to it.  Another option to provide suggestions for product enhancements is to submit an idea through our community: [https://community.tableau.com/s/ideas](url) Providing feedback through the community forums allows other users to vote and discuss feature requests and allows our development team to gauge the demand for each enhancement. Your feedback is invaluable and helps us improve our software.
> > Since the enhancement request has been submitted on your behalf, I will be moving this case into a closed status.

yikes. that is wild. to suggest using "original tabcmd" and then forcing the update for MFA is just piss poor support. I will say my rep was better at least. it took them over a week to simply confirm it is a known issue with no comment on when a fix will be in. this was roughly three weeks ago.

**Comment by til-jmac** [comment]
I have made a workaround Python script to do this, which you can download here. Feedback welcome! https://github.com/TheInformationLab/tableau-tools/tree/main/tableau-server-pdf-downloader

**Comment by BGirkins** [comment]
> This is a huge issue for us as well. Bulk exporting PDFs takes hours when you have to connect to the server for each individual view/PDF. Is there an update on when this functionality will be added? How can we revert to original tabcmd in the meantime?

I do not believe it has been fixed yet. You could likely download the previous tabcmd version and then attempt to reference it instead of the 2.0 one. We simply chose not to because we had already updated all references and syntax changes.

**Comment by aasper-msh** [comment]
> > This is a huge issue for us as well. Bulk exporting PDFs takes hours when you have to connect to the server for each individual view/PDF. Is there an update on when this functionality will be added? How can we revert to original tabcmd in the meantime?
> 
> I do not believe it has been fixed yet. You could likely download the previous tabcmd version and then attempt to reference it instead of the 2.0 one. We simply chose not to because we had already updated all references and syntax changes. also, I believe we were forced into using 2.0 due to the security changes that were pushed around 2.0 launch.

I don't think it's possible for us to use the original tabcmd, as we use Tableau Cloud (not Tableau Server).

**Comment by til-jmac** [comment]
> > > This is a huge issue for us as well. Bulk exporting PDFs takes hours when you have to connect to the server for each individual view/PDF. Is there an update on when this functionality will be added? How can we revert to original tabcmd in the meantime?
> > 
> > 
> > I do not believe it has been fixed yet. You could likely download the previous tabcmd version and then attempt to reference it instead of the 2.0 one. We simply chose not to because we had already updated all references and syntax changes. also, I believe we were forced into using 2.0 due to the security changes that were pushed around 2.0 launch.
> 
> I don't think it's possible for us to use the original tabcmd, as we use Tableau Cloud (not Tableau Server).

There is a way to authenticate with tabcmd 2.0 using PAT and then use the authentication token received with tabcmd 1.0. I can't quite recall the details but I think tabcmd 2.0 saves the auth cookie on the file system and then it's just a case of copying that file to the tabcmd 1.0 directory structure somewhere. Sorry for being so vague but it might be worth investigating this approach if it's critical!

**Comment by til-jmac** [comment]
With the news that tabcmd v1 will be retired for all users pending the move of Tableau Cloud to Hyperforce in October/November, I am hoping this (and other parity issues) are resolved before then, or it will cause quite a few breaking changes to lots of customer workloads. I can see the in-progress tag was added on May 10, but is there any further news on this @jacalata?

**Comment by jacalata** [comment]
It will be added in the next release.

**Comment by eneiner** [comment]
This is not working for me. I just installed and tested this with 2.0.17 with --fullpdf. I no longer see the warning message - but the report is not handling the parameter -blank report. Changing to --pdf get's the output of the report for one view.
